src/canvas/Canvas.ts

```diff
@@ -1334,7 +1334,7 @@ export class Canvas extends SelectableCanvas implements CanvasOptions {
         pointer: this.getPointer(e, true),
         absolutePointer: this.getPointer(e),
       };
-      fireCanvas && this.fire(canvasIn, outOpt);
+      fireCanvas && this.fire(canvasOut, outOpt);
```

Comment:
The fact that TS didn't catch this means probably that somewhere the type is not precise, although I was not able to spot it with a quick check.
Also @ShaMan123 says I ruined TS for him, so I won't stress it 😄

Reply:
Okay I found and fixed the issue on a second look, now it will raise a type error:

<img width="1119" alt="Screenshot 2023-10-03 at 00 25 25" src="https://github.com/fabricjs/fabric.js/assets/10067273/c69af29a-91bf-4adc-9c53-ec995b84e97c">

Reply:
It is not you who ruined TS for me
TS managed by itself
I did look at this part and was confused about it

Reply:
b6f076c is to blame
https://github.com/fabricjs/fabric.js/blob/d55fabcd671b183aa40091c9ed6d4b7fa24c87fa/src/mixins/canvas_events.mixin.ts#L1220-L1227

Reply:
surprising tests did not pick this up, will add

Reply:
they did - the snapshot was wrong